src/d3a/models/myco_matcher/myco_external_matcher.py

```diff
@@ -145,6 +160,8 @@ def _get_validated_recommendations(
         """Return a validated list of BidOfferMatch instances."""
         validated_recommendations = []
         for recommendation in recommendations:
+            if not BidOfferMatch.is_valid_dict(recommended_pair):
+                raise MycoValidationException
```

Comment:
This is the actual task

Reply:
Minor, but how about being more specific in order to know already from the Exception which validation failed?
```suggestion
            if not BidOfferMatch.is_valid_dict(recommended_pair):
                raise MycoValidationException("BidOfferMatch is not valid")
```